ingest/scripts/call_loculus.py

```diff
@@ -218,21 +218,36 @@ def regroup_and_revoke(metadata, sequences, map, config: Config, group_id):
     Submit segments in new sequence groups and revoke segments in old (incorrect) groups in Loculus.
     """
     response = submit_or_revise(metadata, sequences, config, group_id, mode="submit")
-    new_accessions = response[0]["accession"]  # Will be later added as version comment
-
-    url = f"{organism_url(config)}/revoke"
+    new_accessions = {}  # Map from submissionId to new loculus accession
+    for item in response:
+        new_accessions[item["submissionId"]] = item["accession"]
 
     to_revoke = json.load(open(map, encoding="utf-8"))
 
-    loc_values = {loc for seq in to_revoke.values() for loc in seq.keys()}
-    loculus_accessions = set(loc_values)
-
-    accessions = {"accessions": list(loculus_accessions)}
+    old_loculus_keys: dict[
+        str, list[str]
+    ] = {}  # Map from old loculus accession to corresponding new accession(s)
+    for key, value in to_revoke.items():
+        for loc_accession in value:
+            all = old_loculus_keys.get(loc_accession, [])
+            all.append(new_accessions[key])
+            old_loculus_keys[loc_accession] = all
```

Comment:
In my understanding this should not be required because two lines above you will get this actual list, not a copy of it - so this has been assigned all along

Reply:
I think I need the final line because of the case when it was empty, e.g. [] - there it will not be assigned unless I add it back.

Reply:
yes! sorry!